reconstruction.py

```diff
@@ -627,146 +658,169 @@ def reconstruct(self,evrange=(-1,-1,-1)):
 
-                                # If one wants to visualize the new weighted waveforms,
-                                # meaning how much they actual weight for the final average, 
-                                # Ask David for the script changes
+                                        fast_wf_weights_snr = [ (x / max(fast_wf_weights_snr)) for x in fast_wf_weights_snr ]   # Normalization of the weights
+                                        
+                                        for k in range(chs_to_analyse):
+                                        
+                                            for j in range(fast_sampling):
 
-                                if ch == 4:
+                                                weight_average_wf[j] += sing_weig_avg_fast_wf[k][j]*fast_wf_weights_snr[k]/sum(fast_wf_weights_snr)
 
-                                    fast_wf_weights_snr = [ (x / max(fast_wf_weights_snr)) for x in fast_wf_weights_snr ]   # Normalization of the weights
+                                        waveform_info_fast_wei_avg = { 'run' : run, 'event': event, 'channel' : 9, 'trigger' : trg, 'GE' : 9, 'sampling' : "fast"}
 
-                                    for k in range(chs_to_analyse):
+                                        t0_waveforms = time.perf_counter()
+                                        fast_waveform_wei_avg = PMTreco(waveform_info_fast_wei_avg, weight_average_wf, self.pmt_params)
+                                        t1_waveforms = time.perf_counter()
+                                        t_waveforms = t1_waveforms - t0_waveforms
 
-                                        for j in range(fast_sampling):
+                                        self.autotree_pmt_avg.fillPMTVariables_average(fast_waveform_wei_avg)
+                                        self.autotree_pmt_avg.fillTimePMTVariables(t_waveforms)
+                                        #fast_waveform_wei_avg.__repr__()           ## Verbose of averaged waveform
+                                        self.outTree_pmt_avg.fill()
 
-                                            weight_average_wf[j] += sing_weig_avg_fast_wf[k][j]*fast_wf_weights_snr[k]/sum(fast_wf_weights_snr)
+                                        del waveform_info_fast_wei_avg
+                                        del fast_waveform_wei_avg
 
-                                    waveform_info_fast_wei_avg = { 'run' : run, 'event': event, 'channel' : 9, 'trigger' : trg, 'GE' : 9, 'sampling' : "fast"}
+                                    del waveform_info
+                                    del fast_waveform
+                                
+                                # GEM readout. Only available for fast digitizer
+                                # No computing time properties for GEM for now.
+                                if options.include_gem:
```

Comment:
GEMs are saved in the fast digitizer???

Reply:
Yes, *only* in the fast digitizer. I have no opinions on the topic :D

Reply:
ok ok